src/buildercore/fastly/vcl/main.vcl

```diff
@@ -11,8 +11,18 @@ sub vcl_recv {
 sub vcl_fetch {
   #FASTLY fetch
 
-  if ((beresp.status == 500 || beresp.status == 503) && req.restarts < 1 && (req.request == "GET" || req.request == "HEAD")) {
-    restart;
+  if (beresp.status >= 500 && beresp.status < 600) {
+    if (stale.exists) {
+      return(deliver_stale);
+    }
+
+    if (req.restarts < 1 && (req.request == "GET" || req.request == "HEAD")) {
```

Comment:
given the lack of subroutines, the various cases need comments to describe actions like `retry without being logged in`

Reply:
(in a programming language it would be `retry_without_being_logged_in()` which obviates comments)

Reply:
Varnish allows for custom subroutines. Wonder if Fastly does too. 🤔